packages/salesforcedx-utils-vscode/src/telemetry/telemetry.ts

```diff
@@ -0,0 +1,204 @@
+/*
+ * Copyright (c) 2018, salesforce.com, inc.
+ * All rights reserved.
+ * Licensed under the BSD 3-Clause license.
+ * For full license text, see LICENSE.txt file in the repo root or https://opensource.org/licenses/BSD-3-Clause
+ */
+
+import * as path from 'path';
+import * as util from 'util';
+import { ExtensionContext, workspace } from 'vscode';
+import {
+  disableCLITelemetry,
+  isCLITelemetryAllowed
+} from '../cli/cliConfiguration';
+import TelemetryReporter from './telemetryReporter';
+
+const EXTENSION_NAME = 'salesforcedx-vscode-core';
```

Comment:
In future refactor we'll update this so that individual extension just needs to pass in the extension name.